cocos/scripting/js-bindings/manual/ScriptingCore.cpp

```diff
@@ -520,8 +520,8 @@ void ScriptingCore::string_report(JS::HandleValue val) {
 bool ScriptingCore::evalString(const char *string, JS::MutableHandleValue outVal, const char *filename, JSContext* cx, JS::HandleObject global)
 {
     JSAutoCompartment ac(cx, global);
-    JS::PersistentRootedScript script(cx);
-    if (script == nullptr) {
+    JS::RootedScript script(cx);
+    if (&script == nullptr) {
```

Comment:
I think that `script` can't be `nullptr` with the new change as well.
And why did you change `PersistentRootedScript` to `RootedScript`? thanks.